### PR TITLE
Fix for memory leak (value action table)

### PR DESCRIPTION
**Pull request by jakubradtke**
Signed-off-by: Jakub Radtke <jakub.radtke@intel.com>

### CODE REVIEW

File: lib/pmem/ARTree.cpp [hunk 1]
```diff
@@ -293,48 +289,57 @@ void ARTree::Put(const char *key, int32_t keyBytes, const char *value,
...
             // reserved with xreserve
             // pmemobj_defer_free(tree->_pm_pool.get_handle(),(*valPrstPtr.raw_ptr()),&actionsArray[actionsCounter]);
             // pmemobj_free(valPrstPtr.raw_ptr());
-            actionsCounter++;
-            // decrement counter in parent and remove if needed
-            decrementParent(val->parent);
-            /*int status = pmemobj_publish(tree->_pm_pool.get_handle(),
-                                        actionsArray, actionsCounter);*/
+
+            delete[] actionValue;
```
[the remark this answers is not in the document]
Comment (gjerecze) [reply lib/pmem/ARTree.cpp]: I think we need to change delete to delete[] in line 577 as well.

File: lib/pmem/ARTree.cpp [hunk 2]
```diff
@@ -293,48 +289,57 @@ void ARTree::Put(const char *key, int32_t keyBytes, const char *value,
  * */
 void ARTree::decrementParent(persistent_ptr<Node> node) {}
 
+void ARTree::removeFromParent(ValueWrapper *val) {
+    static thread_local struct pobj_action actionsArray[1];
+    static thread_local int actionsCounter = 0;
```
[the remark this answers is not in the document]
Comment (jakubradtke) [reply lib/pmem/ARTree.cpp]: ok,removed and added comment

File: lib/pmem/ARTree.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakubradtke) [reply lib/pmem/ARTree.cpp]: sure

File: lib/pmem/ARTree.h [hunk 3]
```diff
@@ -175,8 +175,14 @@ class ARTree : public DaqDB::RTreeEngine {
     void UpdateValueWrapper(const char *key, uint64_t *ptr, size_t size) final;
     void printKey(const char *key);
     void decrementParent(persistent_ptr<Node> node);
+    void removeFromParent(ValueWrapper *val);
 
   private:
+    inline bool _isValueReservedNotPublished(ValueWrapper *val) {
```
[the remark this answers is not in the document]
Comment (jakubradtke) [reply lib/pmem/ARTree.h]: yep, I know

**Pull request merged**